### PR TITLE
Why-NiJeL Frontend Section

### PR DESCRIPTION
**Pull request by josephbosire**
Fixes the why nijel section of the front end. It seems the html part was commented out.
I made some enhancement to code to make it more simpler and also added the ability to dynamically fetch the menu items for that section.
I also added an angular filter to help slugify urls for better SEO support

How to test?

1. Checkout the branch
 `git checkout origin/whynijel_frontend_section`
2. Run gulp
 `gulp`
3. Navigate to the  why nijel page form the menu section and it should now load the content

### CODE REVIEW

File: app/js/controllers/nav.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const NavCtrl = function ($scope, ClientDataService, $sce) {
+    $scope.trustAsHtml = (template) => {
+        return $sce.trustAsHtml(template);
+    };
+
+    ClientDataService.fetchWhyNijelSections()
+        .then((resp) => {
+            $scope.titles = [];
+            resp.data.sections.forEach(function (section) {
```
Comment (teddyotieno) [inline app/js/controllers/nav.js]: just a nitpick, using arrow functions as opposed to the normal function definitions, for consistency purposes

**Comment by teddyotieno** [comment]
Pulled the branch locally, not sure why but's it's not working for me, a click on the dropdown menu doesn't trigger it's corresponding tab, can only see the url being updated.

File: app/js/controllers/nav.js (hunk 1 above)
Comment (josephbosire) [reply app/js/controllers/nav.js]: Done

**Comment by josephbosire** [comment]
@teddyotieno  Could you try again? Testing it locally seems to be working. Also make sure to clear your browser cache.

**Comment by teddyotieno** [comment]
@judithachieng possible if you pull down this branch and test it out, report on whether the why-nijel drop-downs are connected to the tabs, clear out the site data before you test it out
screen shot 2017-11-10 at 6 28 02 pm png

**Comment by judithachieng** [comment]
@teddyotieno  I have tested it out locally and the drop downs are not connected to the respective tabs.

**Comment by josephbosire** [comment]
@teddyotieno and @judithachieng  Hey guys sorry I had not understood the comment teddy had made before and thought he was saying the page is not loading at all. Totally my fault. Yes that is a bug and the menu don't link to the corresponding tab and will work on a fix for that

**Comment by josephbosire** [comment]
@teddyotieno  and @judithachieng The bug that relates to the navigation action has now been fixed. Selecting a menu option in the why-nijel drop down menu will now link to the appropriate tab

**Comment by teddyotieno** [comment]
Looks good 👍

**Review by teddyotieno — APPROVED** [review]

**Pull request merged**